numba/targets/arraymath.py

```diff
@@ -2344,6 +2344,50 @@ def np_imag_impl(a):
 #----------------------------------------------------------------------------
 # Misc functions
 
+@overload(np.delete)
+def np_delete(arr, obj):
+    # Implementation based on numpy
+    # https://github.com/numpy/numpy/blob/af66e487a57bfd4850f4306e3b85d1dac3c70412/numpy/lib/function_base.py#L4065-L4267
+
+    if not isinstance(arr, (types.Array, types.Sequence)):
+        raise TypingError("delete(): arr must be either an Array or a Sequence")
```

Comment:
This might be considered inconsistent, I'll let @stuartarchibald be the judge of that.

Here is what this type of `TypingError` looks like in other numpy related functions.

```
💥 zsh» git --no-pager grep TypingError numba/targets/arraymath.py
numba/targets/arraymath.py:from numba.errors import RequireLiteralValue, TypingError
numba/targets/arraymath.py:            raise TypingError("Boolean dtype is unsupported (as per NumPy)")
numba/targets/arraymath.py:        raise TypingError('y cannot be a scalar')
numba/targets/arraymath.py:        raise TypingError('y cannot be 0D')
numba/targets/arraymath.py:            raise TypingError('Second argument N must be None or an integer')
numba/targets/arraymath.py:        raise TypingError('shift must be an integer')
numba/targets/arraymath.py:            raise TypingError('xp must be 1D')
numba/targets/arraymath.py:            raise TypingError('fp must be 1D')
numba/targets/arraymath.py:            raise TypingError(complex_dtype_msg)
numba/targets/arraymath.py:                raise TypingError(complex_dtype_msg)
numba/targets/arraymath.py:                raise TypingError(complex_dtype_msg)
numba/targets/arraymath.py:                raise TypingError('ddof must be a real numerical scalar type')
numba/targets/arraymath.py:        raise TypingError(msg)
numba/targets/arraymath.py:            raise TypingError("%s() only supported on 1D arrays " % func_name)
```

Reply:
This also concerns the other raised Exceptions in this PR.